src/main/java/hw6/pages/MetalAndColorsPage.java

```diff
@@ -0,0 +1,214 @@
+
+public class MetalAndColorsPage extends WebPage {
+
+    // Colors dropdown
+    @JDropdown(root = "div[ui=dropdown]",
+            value = ".filter-option",
+            list = "li",
+            expand = ".caret")
+    public Dropdown colorsDropdown;
+
+    public void selectColor(String colorValue) {
+        colorsDropdown.select(colorValue);
+    }
+
+    //Metals dropdown
+    @JDropdown(
+            root = "div[ui=combobox]",
+            value = "input",
+            list = "li",
+            expand = ".caret"
+    )
+    public Dropdown metalsDropdown;
+
+    public void selectMetal(String metalValue) {
+        metalsDropdown.select(metalValue);
+    }
+
+    // Numbers summary
+    @UI("[name=custom_radio_odd]")
+    public RadioButtons oddNumbers;
+
+    @UI("[name=custom_radio_even]")
+    public RadioButtons evenNumbers;
+
+    public void selectNumbers(int[] numbers) {
+        int first;
+        int second;
+        switch (numbers[0]) {
+            case (1):
+                first = 1;
+                break;
+            case (3):
+                first = 2;
+                break;
+            case (5):
+                first = 3;
+                break;
+            default:
+                first = 4;
+                break;
+        }
+        switch (numbers[1]) {
+            case (2):
+                second = 1;
+                break;
+            case (4):
+                second = 2;
+                break;
+            case (6):
+                second = 3;
+                break;
+            default:
+                second = 4;
+                break;
+        }
+
+        oddNumbers.select(first);
+        evenNumbers.select(second);
+    }
+
+
+    //Elements checklist
+    @FindBy(css = "section[id=elements-checklist] input[type=checkbox]")
+    public Checklist earthElements;
+
+    @XPath("//input[@id='g1']")
+    public Checkbox waterCheckBox;
+
+    @XPath("//input[@id='g2']")
+    public Checkbox earthCheckBox;
+
+    @XPath("//input[@id='g3']")
+    public Checkbox windCheckBox;
+
+    @XPath("//input[@id='g4']")
+    public Checkbox fireCheckBox;
+
+    public void selectElements(String[] elements) {
+        cleanUpElements();
+        earthElements.select(elements);
+    }
+
+    public void cleanUpElements() {
+        if (waterCheckBox.isSelected()) {
+            waterCheckBox.uncheck();
+        }
+        if (earthCheckBox.isSelected()) {
+            earthCheckBox.uncheck();
+        }
+        if (windCheckBox.isSelected()) {
+            windCheckBox.uncheck();
+        }
+        if (fireCheckBox.isSelected()) {
+            fireCheckBox.uncheck();
+        }
+    }
+
+    //Vegetables dropdown with checboxes
+    @JDropdown(root = "div[ui=droplist]",
+            value = ".filter-option",
+            list = "li",
+            expand = ".caret")
+    public Dropdown vegetablesDropdown;
+
+    @XPath("//input[@id='g5']")
+    public Checkbox cucumberCheckBox;
```

Comment:
Same story like elements, you already have vegetablesDropdown, this should be enough

Reply:
Removed